src/main/scala/org/downtowndailybread/bethsaida/request/EventRequest.scala

```
@@ -0,0 +1,142 @@
+package org.downtowndailybread.bethsaida.request
+
+import java.sql.{Connection, ResultSet}
+import java.time.{OffsetDateTime, ZonedDateTime}
+import java.util.UUID
+
+import org.downtowndailybread.bethsaida.Settings
+import org.downtowndailybread.bethsaida.exception.event.EventNotFoundException
+import org.downtowndailybread.bethsaida.model.{Event, EventAttribute, HoursOfOperation, InternalUser}
+import org.downtowndailybread.bethsaida.request.util.{BaseRequest, DatabaseRequest}
+import org.downtowndailybread.bethsaida.service.UUIDProvider
+
+class EventRequest(conn: Connection, settings: Settings) extends BaseRequest
+  with DatabaseRequest with UUIDProvider {
+
+  def getAllServiceEvents(serviceId: UUID): Seq[Event] = {
+    getAllEventsInternal(Some(serviceId), None)
+  }
+
+  def getEvent(serviceId: UUID, eventId: UUID): Event = {
+    getAllEventsInternal(Some(serviceId), Some(eventId)) match {
```

What if multiple get returned? Is that not possible?

---

It should throw an event not found exception. Cause it doesn’t know which one to return.

---

It should throw an event not found exception. Cause it doesn’t know which one to return.